### ISSUE TITLE
Parallel method in suite tests not triggering linting errors

### ISSUE THREAD

**Issue by nek1kek** [Antonboom/testifylint#229]
Hello testifying maintainers,

I am encountering an issue where the testifylint does not flag improper use of t.Parallel() within the testify suite structures. According to testify's practice and concurrency model, using t.Parallel() directly within suite-methods should be avoided as it can lead to race conditions or unexpected behaviors in the context of suite test execution. However, the linter does not  report this.

example:
```go
type ParallelRaise struct {
    suite.Suite
}

func TestRunFESAPIUZSuite(t *testing.T) {
    t.Parallel()
    suite.Run(t, new(ParallelRaise))
}

func (s *ParallelRaise) TestReceiptEndpoint() {
    s.T().Parallel()
    s.Run("Test_raise_parallel", func() {
        s.T().Parallel()
        s.Assert().Greater(2, 1)
    })
    s.Run("Test_raise_parallel_v2", func() {
        s.T().Parallel()
        s.Assert().Greater(2, 1)
    })
}

type MySuite struct {
    suite.Suite
}

func (s *MySuite) SetupTest() {
    s.T().Parallel()
}

func (s *MySuite) TestSomething() {
    s.T().Parallel()
    s.Run("1", func() {
        s.T().Parallel()
    })

    s.T().Run("2", func(t *testing.T) {
        t.Parallel()
    })
}
```

**Comment by Antonboom** [comment]
Hi

did you try https://github.com/Antonboom/testifylint?tab=readme-ov-file#suite-subtest-run?

**Comment by nek1kek** [comment]
Yep, the main problem is, the rest of the checks are working fine, but this particular one is not working for our team. Maybe you can show our mistake in calling testifylint.

https://pastebin.com/qFkQYHiF
